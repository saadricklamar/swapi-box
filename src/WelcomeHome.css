.App {
    text-align: center;
  }
  .container {
    display: grid;
    height: 1500px;
    grid-template-columns: 15% 88%;
    grid-template-areas:
    "sidebar header"
    "sidebar main"
    "sidebar main";
  }

  .sidebar {
    grid-area: sidebar;
    width: 100%;
    grid-row: span 6;
    background-color: #41506B ;
    padding-top: 200px;
  }

  header {
    grid-area: header;
    grid-row: span 1;
    background-color: #1B4C9B;
    height: 100px;
    width: 100%;
    font-size: 30px;
    margin: 0;
    padding: 0;
  }

  h1 {
    position: absolute;
    top: 20px;
    left: 55px;
    margin-top: 0;
    font-size: 25px;
    font-family: 'Prompt', sans-serif;
  }

  main {
    grid-area: main;
    grid-row: span 1;
    background-color: #DBE3E6;
    height: 1500px;
    width: 100%;
    display: flex;
    margin: 0;
    background: linear-gradient( rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2) ), url(./images/space.jpg) no-repeat center center fixed;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
  }

  .sidebar button {
    width: 200px;
    height: 100px;
    background-color: #3E619B;
    border: none;
    color: #AFBCC2;
    font-weight: bold;
    text-align: left;
    margin: 0;
    padding: 0 0 0 22px;
    outline: none;
    font-size: 36px;
  }
  
  .sidebar button:hover {
    color: white;
    background-color: #9EC5AB;
    transform: translate(0, 1px);
    cursor: pointer;
  }
  
  .scroll-up {
    overflow: hidden;
    position: relative;
    margin: 0;
   }

   .scroll-up p {
    color: white;
    position: absolute;
    width: 100%;
    height: 80%;
    line-height: 80px;
    font-size: 36px;
    text-align: center;
    /* Starting position */
    -moz-transform:translateY(100%);
    -webkit-transform:translateY(100%);	
    transform:translateY(100%);
    /* Apply animation to this element */	
    -moz-animation: scroll-up 10s linear infinite;
    -webkit-animation: scroll-up 10s linear infinite;
    animation: scroll-up 10s linear infinite;
   }
   /* Move it (define the animation) */
   @-moz-keyframes scroll-up {
    0%   { -moz-transform: translateY(100%); }
    100% { -moz-transform: translateY(-100%); }
   }
   @-webkit-keyframes scroll-up {
    0%   { -webkit-transform: translateY(100%); }
    100% { -webkit-transform: translateY(-100%); }
   }
   @keyframes scroll-up {
    0%   { 
    -moz-transform: translateY(100%); /* Browser bug fix */
    -webkit-transform: translateY(100%); /* Browser bug fix */
    transform: translateY(100%); 		
    }
    100% { 
    -moz-transform: translateY(-100%); /* Browser bug fix */
    -webkit-transform: translateY(-100%); /* Browser bug fix */
    transform: translateY(-100%); 
    }
   }