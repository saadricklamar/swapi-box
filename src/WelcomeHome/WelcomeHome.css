.App {
    text-align: center;
    font-family: 'Orbitron', sans-serif;
}

.container {
    display: grid;
    height: 1800px;
    grid-template-columns: 15% 88%;
    grid-template-areas:
    "sidebar header"
    "sidebar main"
    "sidebar main";
}

.sidebar {
    grid-area: sidebar;
    width: 100%;
    grid-row: span 6;
    background-color:#2e557c;
    padding-top: 200px; 
    
}

header {
    grid-area: header;
    grid-row: span 1;
    background-color: #0097A7;
    height: 100px;
    width: 100%;
    font-size: 30px;
    margin: 0;
    padding: 0;
    border-bottom: 1px inset white;  
    border-left: 1px solid white;
}

h1 {
    position: absolute;
    top: 20px;
    left: 35px;
    margin-top: 0;
    font-size: 30px;
    color: white;
}

main {
    grid-area: main;
    grid-row: span 1;
    background-color: #DBE3E6;
    height: 1800px;
    width: 100%;
    display: flex;
    margin: 0;
    background: linear-gradient( rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2) ), url(../images/space.jpg) no-repeat center center fixed;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
    border-left: 1px solid white;
}

.sidebar button {
    width: 200px;
    height: 100px;
    background-color: #0097A7;
    border: none;
    color: white;
    font-weight: bold;
    text-align: left;
    margin: 0;
    padding: 0 0 0 22px;
    outline: none;
    font-size: 30px;
    font-family: 'Orbitron', sans-serif;
    border-bottom: 1px solid white;
}
  
.sidebar button:hover {
    color: white;
    background-color: orange;
    transform: translate(0, 1px);
    cursor: pointer;
}

.sidebar button:focus {
    background:  #41D3BD;
  -webkit-box-shadow: inset 0px 0px 5px #c1c1c1;
     -moz-box-shadow: inset 0px 0px 5px #c1c1c1;
          box-shadow: inset 0px 0px 5px #c1c1c1;
   outline: none;
   border-left: 5px solid white;
}

header button {
    width: 220px;
    height: 50px;
    background-color: #2e557c;
    border: none;
    color: white;
    font-weight: bold;
    text-align: left;
    margin: 25px 0 0 1000px;
    padding: 0 0 0 22px;
    outline: none;
    font-size: 18px;
    font-family: 'Orbitron', sans-serif;
    border: 1px solid white;
}

header button:hover {
    color: white;
    background-color: #9EC5AB;
    background-color: orange;
    transform: translate(0, 1px);
    cursor: pointer;
}

header button:focus {
    background: #41D3BD;
  -webkit-box-shadow: inset 0px 0px 5px #c1c1c1;
     -moz-box-shadow: inset 0px 0px 5px #c1c1c1;
          box-shadow: inset 0px 0px 5px #c1c1c1;
   outline: none;
}
  
.scroll-up {
    overflow: hidden;
    position: relative;
    margin: 0;
}

.scroll-up p {
    color: white;
    position: absolute;
    width: 100%;
    height: 80%;
    line-height: 80px;
    font-size: 36px;
    text-align: center;
    /* Starting position */
    -moz-transform:translateY(100%);
    -webkit-transform:translateY(100%);	
    transform:translateY(100%);
    /* Apply animation to this element */	
    -moz-animation: scroll-up 10s linear infinite;
    -webkit-animation: scroll-up 10s linear infinite;
    animation: scroll-up 10s linear infinite;
   }
   /* Move it (define the animation) */
   @-moz-keyframes scroll-up {
    0%   { -moz-transform: translateY(100%); }
    100% { -moz-transform: translateY(-100%); }
   }
   @-webkit-keyframes scroll-up {
    0%   { -webkit-transform: translateY(100%); }
    100% { -webkit-transform: translateY(-100%); }
   }
   @keyframes scroll-up {
    0%   { 
    -moz-transform: translateY(100%); /* Browser bug fix */
    -webkit-transform: translateY(100%); /* Browser bug fix */
    transform: translateY(100%); 		
    }
    100% { 
    -moz-transform: translateY(-100%); /* Browser bug fix */
    -webkit-transform: translateY(-100%); /* Browser bug fix */
    transform: translateY(-100%); 
    }
}

.fa-galactic-senate {
     position: absolute;
     top: 80px;
     left: 80px;
     font-size: 86px;
     color: orange;
}
   