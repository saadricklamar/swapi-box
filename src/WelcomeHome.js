import React, { Component } from 'react';
import './WelcomeHome.css';


class WelcomeHome extends Component {
  constructor(props) {
    super(props)
    this.state = {
   
    }
  }

  render() {
    return (
    <div className="App">
     <section className='container'>
      <section className='sidebar'>
         <h1>SWAPIBox</h1>
         <button value='people' onClick={this.props.display}>People</button>
         <button value='planets' onClick={this.props.display}>Planets</button>
         <button value='vehicles' onClick={this.props.display}>Vehicles</button>
         <button>Films</button>
         <button>Species</button>
         <button>Starships</button>
      </section>
      <section>
        <header>
          
        </header>
        <main className='scroll-up'>
          <p>{this.props.openingCrawl}</p>
        </main>
      </section>
     </section>
    </div>
    );
  }
}

export default WelcomeHome;