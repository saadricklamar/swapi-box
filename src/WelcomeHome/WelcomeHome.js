import React, { Component } from 'react';
import './WelcomeHome.css';

const WelcomeHome = (props) => {
    return (
    <div className="App">
     <section className='container'>
      <section className='sidebar'>
         <h1>SWAPIBox</h1>
         <i class="fab fa-galactic-senate"></i>
         <button value='people' onClick={props.displayPeople} class='active'>People</button>
         <button value='planets' onClick={props.displayPlanets}>Planets</button>
         <button value='vehicles' onClick={props.displayVehicles}>Vehicles</button>
         <button value='films' onClick={props.displayFilms}>Films</button>
         <button>Species</button>
         <button>Starships</button>
      </section>
      <section>
        <header>
          <div className='scroll-right'>
            <button>View <span>0</span> Favorites </button>
          </div>
        </header>
        <main className='scroll-up'>
          <p>{props.openingCrawl}</p>
        </main>
      </section>
     </section>
    </div>
    );
}

export default WelcomeHome;