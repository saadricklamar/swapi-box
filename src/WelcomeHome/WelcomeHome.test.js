import React from 'react';
import WelcomeHome from './WelcomeHome.js';
import { shallow } from 'enzyme';

const mockFunc = jest.fn();
const mockOpeningCrawl = 'It is a period of civil war.Rebel spaceships, striking from a hidden base, have won their first victory against the evil Galactic Empire. During the battle, Rebel spies managed to steal secret plans to the Empire\'s ultimate weapon, the DEATHSTAR, an armored space station with enough power to destroy an entire planet.Pursued by the Empire\'s sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxy.... A New Hope 1977-05-25';

describe('WelcomeHome', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<WelcomeHome
                    openingCrawl={mockOpeningCrawl}
                    display={mockFunc}
                      />)
});
it('Should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
});
