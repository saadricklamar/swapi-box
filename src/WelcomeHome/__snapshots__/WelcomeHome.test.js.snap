// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeHome Should match the snapshot 1`] = `
<div
  className="App"
>
  <section
    className="container"
  >
    <section
      className="sidebar"
    >
      <h1>
        SWAPIBox
      </h1>
      <i
        class="fab fa-galactic-senate"
      />
      <button
        class="active"
        onClick={[MockFunction]}
        value="people"
      >
        People
      </button>
      <button
        onClick={[MockFunction]}
        value="planets"
      >
        Planets
      </button>
      <button
        onClick={[MockFunction]}
        value="vehicles"
      >
        Vehicles
      </button>
      <button>
        Films
      </button>
      <button>
        Species
      </button>
      <button>
        Starships
      </button>
    </section>
    <section>
      <header>
        <div
          className="scroll-right"
        >
          <p>
            May the force be with you...
          </p>
          <button>
            View 
            <span />
             Favorites  
            <i
              class="far fa-heart"
            />
          </button>
        </div>
      </header>
      <main
        className="scroll-up"
      >
        <p>
          It is a period of civil war.Rebel spaceships, striking from a hidden base, have won their first victory against the evil Galactic Empire. During the battle, Rebel spies managed to steal secret plans to the Empire's ultimate weapon, the DEATHSTAR, an armored space station with enough power to destroy an entire planet.Pursued by the Empire's sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxy.... A New Hope 1977-05-25
        </p>
      </main>
    </section>
  </section>
</div>
`;
